lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -31,67 +32,77 @@ type Props = {
  * It maps AuthenticationContext to SessionContext
  */
 const SessionAuth: React.FC<Props> = ({ children, ...props }) => {
-    const [sessionContext, setSessionContext] = useState<SessionContextType>();
+    const [context, setContext] = useState<SessionContextType>();
+
     const session = Session.getInstanceOrThrow();
 
-    const buildSessionContext = async (): Promise<SessionContextType> => {
-        const sessionExists = await session.doesSessionExist();
+    const setInitialContext = async (): Promise<void> => {
+        const sessionExists = await doesSessionExist();
 
-        if (!sessionExists) {
-            return {
+        if (sessionExists === false) {
+            return setContext({
                 doesSessionExist: sessionExists,
                 jwtPayload: {},
                 userId: "",
-            };
+            });
         }
 
-        const [jwtPayload, userId] = await Promise.all([session.getJWTPayloadSecurely(), session.getUserId()]);
+        const [jwtPayload, userId] = await Promise.all([getJWTPayloadSecurely(), getUserId()]);
 
-        return {
+        return setContext({
             doesSessionExist: sessionExists,
             jwtPayload,
             userId,
-        };
-    };
-
-    const updateSessionContext = async () => {
-        const sessionContext = await buildSessionContext();
-
-        setSessionContext(sessionContext);
+        });
     };
 
-    const onHandleEvent = async (ctx: OnHandleEventContext) => {
-        switch (ctx.action) {
+    const onHandleEvent = async (event: RecipeEvent) => {
+        switch (event.action) {
+            case "SESSION_CREATED":
+                return setContext({
+                    doesSessionExist: true,
+                    userId: await getUserId(),
+                    jwtPayload: await getJWTPayloadSecurely(),
+                });
             case "REFRESH_SESSION":
+                return setContext({
+                    doesSessionExist: true,
+                    userId: context === undefined ? "" : context.userId,
+                    jwtPayload: await getJWTPayloadSecurely(),
+                });
             case "SIGN_OUT":
             case "UNAUTHORISED":
-                return updateSessionContext();
+                return setContext({
```

Comment:
Setting this here is actually a breaking change since previously, users had to redirect to auth page after sign out or session expiry manually.

Reply:
I think it makes sense to make this change and bump the version appropriately.

If we have those automatic context updates, then if a user is using `SessionAuth` (indirectly, through the library) the state is consistent, because we do not allow `SessionAuth` to render its children if there's no session, which is one concern of this component.

What do you think?